src/main/java/com/github/xetorthio/jedisque/BinaryJedisque.java

```diff
@@ -0,0 +1,62 @@
+package com.github.xetorthio.jedisque;
+
+import redis.clients.jedis.Protocol;
+import redis.clients.jedis.exceptions.JedisConnectionException;
+import redis.clients.util.SafeEncoder;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.*;
+
+public class BinaryJedisque extends redis.clients.jedis.Connection {
+	static final private int DISQUE_PORT = 7711;
+	private final List<URI> uris = new ArrayList<URI>();
+	private Random randomGenerator = new Random();
+
+	public BinaryJedisque() {
+		try {
+			uris.add(new URI("disque://localhost:" + DISQUE_PORT));
+		} catch (URISyntaxException e) {
+		}
+	}
+
+	public BinaryJedisque(URI... uris) {
+		this.uris.addAll(Arrays.asList(uris));
+	}
+
+	@Override
+	public void connect() {
+		while (!this.isConnected()) {
```

Comment:
Yes. A failed node may get back to a running state (e.g., by restarting the same instance, assigning the hostname of the failing node to a new instance, etc.). Hence, retrying failed nodes allow the application/jedisque to get back to regular behavior without restarting the application.